cli/ballerina-cli-utils/src/main/java/org/ballerinalang/cli/utils/BVMEmbeddedExecutor.java

```diff
@@ -54,12 +56,17 @@ public void execute(String programArg, boolean isFunction, String... args) {
         if (resource == null) {
             throw new BLangCompilerException("Missing internal modules when retrieving remote package");
         }
+        
         try {
             URI balxResource = resource.toURI();
-            // TODO: 8/5/18 return BValue[] ignored 
-            ExecutorUtils.execute(balxResource, isFunction, isFunction ? functionName : null, args);
+            BValue[] returns = ExecutorUtils.execute(balxResource, isFunction, isFunction ? functionName : null, args);
+            if (returns.length == 1 && returns[0] instanceof BInteger) {
+                return ((BInteger) returns[0]).intValue();
+            } else {
+                return 0L;
+            }
```

Comment:
This function can handle the error here, or can return a java Optional of errors.

Reply:
We cannot return error here. As error is in the core and not in ballerina-lang. Cannot import the core into this module.